pkg/cloud/buckets/interface.go

```diff
@@ -1,8 +1,13 @@
 package buckets
 
+import "bufio"
+
 // Provider represents a bucket provider
+//go:generate pegomock generate github.com/jenkins-x/jx/pkg/cloud/buckets Provider -o mocks/buckets_interface.go
 type Provider interface {
 	// CreateNewBucketForCluster creates a new dynamically named bucket
 	CreateNewBucketForCluster(clusterName string, bucketKind string) (string, error)
 	EnsureBucketIsCreated(bucketURL string) error
+	UploadFileToBucket(bytes []byte, outputName string, bucketURL string) (string, error)
```

Comment:
I think using a `Reader` interface might be more scalable than `[]bytes`.

Reply:
Good idea, just changed it.